File: 917-boats-to-save-people/boats-to-save-people.cpp
class Solution {
public:
    int numRescueBoats(vector<int>& people, int limit) {
        sort(people.begin(), people.end());

        int i = 0, j = people.size()-1;
        int boats = 0;
        while (i <= j){
            int weight = people[i] + people[j];
            if (weight > limit){
                j-=1;
            } else {
                i+=1;
                j-=1;
            }
            boats+=1;
        }
        return boats;
    }
};

File: 917-boats-to-save-people/boats-to-save-people.go
func numRescueBoats(people []int, limit int) int {
    sort.Ints(people)
    boats := 0
    l, r := 0, len(people)-1

    for l <= r {
        sum := people[l] + people[r]

        if sum <= limit {
            boats++
            l++
            r--
        } else {
            boats++
            r--
        }
    }

    return boats
}